clients/src/main/java/org/apache/kafka/clients/admin/internals/AllBrokersStrategy.java

```diff
@@ -0,0 +1,212 @@
+import java.util.Map;
+import java.util.Objects;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class is used for use cases which require requests to be sent to all
+ * brokers in the cluster.
+ *
+ * This is a slightly degenerate case of a lookup strategy in the sense that
+ * the broker IDs are used as both the keys and values. Also, unlike
+ * {@link CoordinatorStrategy} and {@link PartitionLeaderStrategy}, we do not
+ * know the set of keys ahead of time: we require the initial lookup in order
+ * to discover what the broker IDs are. This is represented with a more complex
+ * type {@code Future<Map<Integer, Future<V>>} in the admin API result type.
+ * For example, see {@link org.apache.kafka.clients.admin.ListTransactionsResult}.
+ */
+public class AllBrokersStrategy implements AdminApiLookupStrategy<AllBrokersStrategy.BrokerKey> {
+    public static final BrokerKey ANY_BROKER = new BrokerKey(OptionalInt.empty());
+    public static final Set<BrokerKey> LOOKUP_KEYS = Collections.singleton(ANY_BROKER);
+    private static final ApiRequestScope SINGLE_REQUEST_SCOPE = new ApiRequestScope() {
+    };
+
+    private final Logger log;
+
+    public AllBrokersStrategy(
+        LogContext logContext
+    ) {
+        this.log = logContext.logger(AllBrokersStrategy.class);
+    }
+
+    @Override
+    public ApiRequestScope lookupScope(BrokerKey key) {
+        return SINGLE_REQUEST_SCOPE;
+    }
+
+    @Override
+    public MetadataRequest.Builder buildRequest(Set<BrokerKey> keys) {
+        validateLookupKeys(keys);
+        // Send empty `Metadata` request. We are only interested in the brokers from the response
+        return new MetadataRequest.Builder(new MetadataRequestData());
+    }
+
+    @Override
+    public LookupResult<BrokerKey> handleResponse(Set<BrokerKey> keys, AbstractResponse abstractResponse) {
+        validateLookupKeys(keys);
+
+        MetadataResponse response = (MetadataResponse) abstractResponse;
+        MetadataResponseData.MetadataResponseBrokerCollection brokers = response.data().brokers();
+
+        if (brokers.isEmpty()) {
+            log.debug("Metadata response contained no brokers. Will backoff and retry");
+            return LookupResult.empty();
+        } else {
+            log.debug("Discovered all brokers {} to send requests to", brokers);
+        }
+
+        Map<BrokerKey, Integer> brokerKeys = brokers.stream().collect(Collectors.toMap(
+            broker -> new BrokerKey(OptionalInt.of(broker.nodeId())),
+            MetadataResponseData.MetadataResponseBroker::nodeId
+        ));
+
+        return new LookupResult<>(
+            Collections.singletonList(ANY_BROKER),
+            Collections.emptyMap(),
+            brokerKeys
+        );
+    }
+
+    private void validateLookupKeys(Set<BrokerKey> keys) {
+        if (keys.size() != 1) {
+            throw new IllegalArgumentException("Unexpected key set " + keys);
```

Comment:
nit: I would add a `:` after `set` to stay consistent with the other error messages in this PR.